documentation/swagger-good-patterns.md

```diff
@@ -13,10 +13,18 @@
  - Checks on the Resource Model
     - Every "Resource" Model should have "location" as a "required" property as per the mandate from ARM team; no exceptions are allowed over here.
     - Every "Resource" Model should be tagged with "x-ms-azure-resource". This will cause the Resource to inherit from the Resource definition in the client runtime.
+    - GET/PUT/PATCH for a resource should return the same "Resource" Model, also PUT should accept the same body (there can be readonly, optional, required properties).
  - Consistency in the naming of list methods
```

Comment:
For PATCH I have doubts what we should do, structure wise the Resource Body is good, but all values should be nullable, so for int it does not matter that whether in swagger there is any default value or not, or whether that is  optional or not. I think for PATCH code generation should take care of generating the same structure as of Resource body but with nullable types. For Example if a Resource is like { poolQuota: int, size: string }, structure generated for PUT request/response, GET response, PATCH response should be { int PoolQuota, string Size} where as for PATCH request it should be { int? PoolQuota, string Size }. In swagger for PATCH request same body should be there as for PUT.

I am not sure for this pull request we should make any updates on PATCH.